### DISCUSSION TITLE
Tagging with Azure registry

### DISCUSSION THREAD

**Discussion by diberry** [microsoft/vscode-docker#3152] [Q&A]
If I want to set up deployment from my ACR container to Azure app service, do I need to tag with the ACR every time or does the extension know now that I've tagged it with ACR, it remembers that? 

The reason I ask is that I thought on redeployment (local changes in VSCode, then local build, then push) didn't require retagging but that isn't working for me now.

**Comment by bwateratmsft** [comment]
You can probably accomplish what you're looking for by changing what tag is applied after you build your image, so that the image builds directly into the ACR-tagged version, and doesn't require a retag. In the extension if you push an image with a tag that matches a connected registry (including ACR), it shouldn't prompt again for tagging (if it does, that's a bug).

I'd suggest changing your Docker Build command to something like this:
```shell
docker build -t myacr.azurecr.io/myimage:latest -f Dockerfile .
```

If you're using a `docker-build` task from the extension, you can use the `"tag"` property:
```json
{
	"type": "docker-build",
	"label": "docker-build",
	"platform": "node",
	"dockerBuild": {
		"dockerfile": "${workspaceFolder}/Dockerfile",
		"context": "${workspaceFolder}",
		"pull": true,
		"tag": "myacr.azurecr.io/myimage:latest"
	}
},
```

**Comment by diberry** [comment]
Followup question then....

My usual pattern (correct me if you see where I'm going off base) is to build a local image (image A: latest), then tag it such as (acr/image A:latest.

When I make a change and want to rebuild, am I rebuilding imageA or acr/image A? My habit is to rebuild image A then retag to acr, then push acr but that seems like more steps than necessary. 

I'm looking for best practices, customer recommendations.

**Reply by bwateratmsft** [reply]
> When I make a change and want to rebuild, am I rebuilding imageA or acr/imageA?

Most likely you're rebuilding imageA but it depends what's in your `docker build` command. The approach of rebuilding imageA, retagging to acr/imageA, and then pushing is valid, just has an extra step.

Personally I don't see any issue with directly building as acr/imageA to save the tagging step.

**Reply by diberry** [reply]
Ok, awesome to know. Another question - this is for docs, so still hoping for best practices...I wouldn't tag with latest, I would ideally tag with a version each time, correct? Then tag the most recent version with lastest. And then push both version and latest?

**Reply by bwateratmsft** [reply]
Yeah, that approach sounds reasonable to me. That said, I'm an extension author, I don't run any containerized services :smile:
